### ISSUE TITLE
Update events resolver to use store ordering

### ISSUE THREAD

**Issue by jamesdphillips** [sensu/sensu-go#4487]
After https://github.com/sensu/sensu-go/issues/4485 has been implemented, we'll want to update the events GraphQL resolver. I believe the steps to be as follows:

1. Since the etcd store will not be implementing ordering and filtering, I think we should avoid using the dataloader in the events field resolver. This is the because the data loader implicitly caches the results from the store for the duration of the session and differences between the store implementations may lead to subtle bugs. As such I think the resolver should be updated to directly call the loader's [listEvents()](https://github.com/sensu/sensu-go/blob/main/backend/apid/graphql/dataloader.go#L148-L168) function instead.
1. In the field resolver we'll want to lift the order and offset into the context before making a call to the [listEvents()](https://github.com/sensu/sensu-go/blob/main/backend/apid/graphql/dataloader.go#L148-L168) function.
1. In the field resolver we'll want to check if the [events store supports filtering](https://sumologic.slack.com/archives/C024XK35Z3Q/p1637694698190400) and if so make a call to the store's `EventsCount` method, we can use the result we can set the [totalCount](https://github.com/sensu/sensu-go/blob/main/backend/apid/graphql/namespace.go#L291) field.

Voila!

**Comment by jamesdphillips** [comment]
It occurred to me that there is a slight wrinkle in this plan... The GraphQL resolver's signature expects pagination to be offset based while the store API is cursor based; if we actually want to get away from scanning the entire events keyspace / table we'll need to address this. Unfortunately due to federation we'll need to do this is a way that doesn't break the existing API.

A proposed solution:

* soft deprecate the event resolvers `offset` argument; any values provider are ignored.
* add a new `after` argument to the revolvers that will be equivalent to the continue token.
* update the resolvers response such have the `totalCount` field always be zero.
* add a `endCursor` field to the `OffsetPageInfo` type; we will fill this value with the continue token. (We could also duplicate the `OffsetPageInfo` type creating a new type named `EventConnectionPageInfo`, mark the `offset` field as deprecated, and then also add the `endCursor` field. This way we avoid polluting the existing type.)
* update the frontend to select and use the `endCursor`

```graphql
type Namespace {
  # ...
  
  "List events associated with the namespace."
  events(
    # moving forward we'll ignore any value given to the 'offset' arg
    offset: Int = 0 @deprecated(reason: "please use the after argument instead"),
    # proposed new field; equivalent to the "continue" token
    after: String,
    "Limit adds optional limit to the number of entries returned."
    limit: Int = 10,
    "OrderBy adds optional order to the records retrieved."
    orderBy: EventsListOrder = SEVERITY
    "Filters ..."
    filters: [String!] = [],
  ): EventConnection!
}

# ...

"A connection to a sequence of records."
type EventConnection {
  nodes: [Event!]!
  # s/OffsetPageInfo/EventConnectionPageInfo/
  pageInfo: EventConnectionPageInfo!
}

# ...

"""
Information about the current page.
"""
type EventConnectionPageInfo {
  "When paginating forward, are there more items?"
  hasNextPage: Boolean!

  "When paginating backward, are there more items?"
  hasPreviousPage: Boolean!

  "Next cursor to use when paginating forward; null if there are no more items."
  endCursor: String # new

  "Next offset to use when paginating forward; null if there are no more items."
  nextOffset: Int @deprecated(reason: "select endCursor instead")

  "Previous offset to use when paginating backward; null if at terminal."
  previousOffset: Int @deprecated(reason: "always null; no going backward")

  "Total count of records in relationship."
  totalCount: Int! @deprecated(reason: "always zero")
}
```

**Drawbacks**

The main drawback to this approach is that in a federated environment when a gateway cluster's version is _less than_ 6.6.x and is communicating with a cluster _greater or equal to_ 6.6.x it will appear as if there is only one page, even if there are more items than the given offset. This is unfortunately but I think reduced complexity and that the greatly reduced memory usage is preferable.

**Comment by portertech** [comment]
Shipped in 6.6
